### ISSUE TITLE
Lack of boundary checking

### ISSUE THREAD

**Issue by sobomax** [sipcapture/hep-wireshark#9]
Hi, we've tried to use this dissector to debug new implementation of the HEPv3 client, but it seems to be lacking any even basic boundary/sanity checks, causing wireshark to hang solid while dissecting small capture (attached). This severely limits its usefulness for such use cases as well could lead to problems while processing large number of real-world packets, as wire errors do happen in real life.

[cap.zip](https://github.com/sipcapture/hep-wireshark/files/2078080/cap.zip)

**Comment by lmangani** [comment]
Hey @sobomax what version of Wireshark are you using?

**Comment by sobomax** [comment]
Latest for windows I think. 
![screenshot749](https://user-images.githubusercontent.com/218662/41063873-c6bf009c-698e-11e8-9f3b-d01fc8b66cc5.png)

**Comment by sobomax** [comment]
P.S. I looked at the code briefly and I don't see any validation of the length field(s). Those values just taken verbatim, never compared to the actual packet size. Which is likely to be the cause.

**Comment by lmangani** [comment]
I agree and that's an excellent observation, we'll do our best to sort this out, just wanted to make sure it was not platform specific - so far personally had no such issues and used it pretty intensively on Linux.
